Fix currency filter and country filter

zackt25 opened the pull request:
This will enable the multi selection for filtering in:

`currency` in the Contracts view
`country` in the Vendors view

Before you were only able to filter on 1 selection in regards to currency and country. Now you can select multiple and it will filter out.

This will close #187

itdependsnetworks commented on nautobot_device_lifecycle_mgmt/forms.py:
```diff
@@ -913,11 +919,7 @@ class ProviderLCMFilterForm(BootstrapMixin, forms.ModelForm):
-        required=False,
-        choices=add_blank_choice(CountryCodes.CHOICES),
-    )
+    country = forms.MultipleChoiceField(required=False, choices=CountryCodes.CHOICES, widget=StaticSelect2Multiple())
```
why remove `add_blank_choice`?

zackt25 replied:
I copied and pasted something with fixed and that wasn't in there. I added it back.

progala replied:
@itdependsnetworks I don't think `add_blank_choice` does anything when used with `MultipleChoiceField`? We previously had `ChoiceField` here hence why `add_blank_choice` was used.

zackt25 replied:
I tested both ways and there isn't a noticeable difference when selecting the filter.

itdependsnetworks replied:
ok, makes sense, yet another pylint nautobot use case to consider.

zackt25 commented:
It was agreed from now on for bug fixes we will support python 3.8 and 3.9. I've removed 3.7 from the CI file and changed the toml file to be higher than 3.8.
